test/test_enzyme.jl

```diff
@@ -23,3 +88,78 @@ f(grid) = CenterField(grid)
 
     @test size(primal) == size(shadow)
 end
+
+
+@testset "Enzyme on advection and diffusion" begin
+    Nx = Ny = 64
+    Nz = 8
+
+    Lx = Ly = L = 2π
+    Lz = 1
+
+    x = y = (-L/2, L/2)
+    z = (-Lz/2, Lz/2)
+    topology = (Periodic, Periodic, Bounded)
+
+    grid = RectilinearGrid(size=(Nx, Ny, Nz); x, y, z, topology)
+    diffusion = VerticalScalarDiffusivity(κ=0.1)
+
+    u = XFaceField(grid)
+    v = YFaceField(grid)
+
+    U = 1
+    u₀(x, y, z) = - U * cos(x + L/8) * sin(y) * (z + L/2)
+    v₀(x, y, z) = + U * sin(x + L/8) * cos(y) * (z + L/2)
+
+    set!(u, u₀)
+    set!(v, v₀)
+    fill_halo_regions!(u)
+    fill_halo_regions!(v)
+
+    @inline function tracer_flux(x, y, t, c, p)
+        c₀ = p.surface_tracer_concentration
+        u★ = p.piston_velocity
+        return - u★ * (c₀ - c)
+    end
+
+    parameters = (surface_tracer_concentration = 1,
+                  piston_velocity = 0.1)
+
+    top_c_bc = FluxBoundaryCondition(tracer_flux, field_dependencies=:c; parameters)
+    c_bcs = FieldBoundaryConditions(top=top_c_bc)
+
+    # TODO:
+    # 1. Make the velocity fields evolve
+    # 2. Add surface fluxes
+    # 3. Do a problem where we invert for the tracer fluxes (maybe with CATKE)
+
+    model = HydrostaticFreeSurfaceModel(; grid,
+                                        tracer_advection = WENO(),
+                                        tracers = :c,
+                                        buoyancy = nothing,
+                                        velocities = PrescribedVelocityFields(; u, v),
+                                        closure = diffusion)
+
+    # Compute derivative by hand
+    κ₁, κ₂ = 0.9, 1.1
+    c²₁ = stable_diffusion!(model, 1, κ₁)
+    c²₂ = stable_diffusion!(model, 1, κ₂)
+    dc²_dκ_fd = (c²₂ - c²₁) / (κ₂ - κ₁)
+
+    # Now for real
+    amplitude = 1.0
+    κ = 1.0
+    dmodel = Enzyme.make_zero(model)
+    set_diffusivity!(dmodel, 0)
+
+    dc²_dκ = autodiff(Enzyme.Reverse,
+                      stable_diffusion!,
+                      Duplicated(model, dmodel),
+                      Const(amplitude),
+                      Active(κ))
+
+    @info """ \n
+    Enzyme computed $dc²_dκ
+    Finite differences computed $dc²_dκ_fd
+    """
```

Comment:
This doesn't seem like a test. What is it tested here? Where is the `@test`?

Reply:
The print statement checks that the Enzyme-computed derivative and FD approximation are close. I just added an @test statement to check this too, making sure the relative error is within `tol = 0.01` (we can decide on another `tol` value).

Reply:
We usually don't need to print anything for the purpose of the test --- all we care about is the green check mark for CI.